src/renderer/components/composer/menuAttachment.tsx

```diff
@@ -101,6 +105,36 @@ export default function MenuAttachment({
     tx,
   ])
 
+  const selectContact = async () => {
+    const { setLastPath } = rememberLastUsedPath(LastUsedSlot.Attachment)
+    let dialogId = ''
+    /**
+     * TODO: reduce the overhead: just provide a vcardContact to draft/message
+     * and send it as a message. No need to get the vcard from core to create
+     * a tmp file to attach it as a file which is then converted into a vcardContact again
+     */
+    const addContactAsVcard = async (selectedContact: T.Contact) => {
+      if (selectedContact) {
+        const vCardContact = await BackendRemote.rpc.makeVcard(
+          selectedAccountId(),
+          [selectedContact.id]
+        )
+        const cleanDisplayname = selectedContact.displayName.replace(
+          /[^a-z_A-Z0-9]/gi,
+          ''
+        )
+        const tmp_file = await runtime.writeTempFileFromBase64(
+          `VCard-${cleanDisplayname}.vcf`,
+          btoa(vCardContact)
```

Comment:
`btoa` does not work with emojis.

This method works better:
```
{ var reader = new FileReader(); reader.readAsDataURL(new Blob([await exp.rpc.makeVcard(120, [10])])); reader.onloadend = () => console.log(reader.result) }
```

Reply:
This doesn't work. Any solution trying to convert utf8 to base64 does that by doing an additional encoding which needs to be decoded again after converting it back from base64.
So if I get it right, the solution in https://github.com/deltachat/deltachat-core-rust/pull/5677 does not help, since composer recreates the draft message without keeping the message id?
So I suggest to implement a writeTempFile similar to writeTempFileFromBase64 and remove that again onve Composer is refactored

Reply:
I guess it would work if we would always use blob/buffer instead of atob & btob, but file also works